### ISSUE TITLE
oauth2-proxy returns 500 when a broken URL is being requested

### ISSUE THREAD

**Issue by m0ppers** [oauth2-proxy/oauth2-proxy#1516]
we are using `--skip-provider-button=true` and when external requests contain broken querystrings the proxy would return status code 500 and not a 400 (like I would expect?).

Example: /?q=%vaeritas%&va=b&t=hc&ia=web (%va is not a valid Querystring)

## Expected Behavior

oauth2-proxy should throw a HTTP 400 (as the request is gibberish)

## Current Behavior

a 500 is being thrown.

## Possible Solution

```
diff --git a/oauthproxy.go b/oauthproxy.go
index 69bcdbc..bafa756 100644
--- a/oauthproxy.go
+++ b/oauthproxy.go
@@ -677,7 +677,7 @@ func (p *OAuthProxy) OAuthStart(rw http.ResponseWriter, req *http.Request) {
        appRedirect, err := p.appDirector.GetRedirect(req)
        if err != nil {
                logger.Errorf("Error obtaining application redirect: %v", err)
-               p.ErrorPage(rw, req, http.StatusInternalServerError, err.Error())
+               p.ErrorPage(rw, req, http.StatusBadRequest, err.Error())
                return
        }

```

This is a potential fix but my guess is that this is a bit naive?

## Context

A 500 in our logs made us start an investigation whereas a 400 would just have been ignored by us :tea: 

## Your Environment

Docker container

- Version used:
v7.2.1

**Comment by JoelSpeed** [comment]
Is there anything else being logged that explains why this is an error? GetRedirect shouldn't be calling any external resource so this must be something internal

From what I can see this is still a valid 500 error, though it's due to user configuration I don't think sending a bad request to the user makes sense, that would imply the request they made was bad, which it wasn't

**Comment by m0ppers** [comment]
"due to user configuration" implies that something is misconfigured and in that case a 500 would be justified. However I don't see how that is the case here? The configuration is working as expected but when people craft invalid URLs and send it to the proxy a 500 is happening.

**Comment by JoelSpeed** [comment]
I'm not sure now what I was talking about in my last message. Having re-read the issue, I see what you mean, if the request input is gibberish, we should be returning a 400, I agree

**Comment by anjakammer** [comment]
This issue is still relevant.

**Comment by Niksko** [comment]
Hey @JoelSpeed, I've opened a fix for this, would be great if this fix could make it in to 1.8.0
